src/bridge/views/Login/Mnemonic.js

```diff
@@ -0,0 +1,160 @@
+import { Just, Nothing } from 'folktale/maybe';
+import React from 'react';
+import { Button } from '../../components/Base';
+import {
+  Input,
+  MnemonicInput,
+  InnerLabel,
+  InputCaption,
+} from '../../components/Base';
+import { Row, Col } from '../../components/Base';
+
+import { DEFAULT_HD_PATH, walletFromMnemonic } from '../../lib/wallet';
+import { compose } from '../../lib/lib';
+import { withWallet } from '../../store/wallet';
+
+class Mnemonic extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      mnemonic: '',
+      passphrase: '',
+      hdpath: DEFAULT_HD_PATH,
+    };
+
+    this.handleMnemonicInput = this.handleMnemonicInput.bind(this);
+    this.handlePassphraseInput = this.handlePassphraseInput.bind(this);
+    this.handleHdPathInput = this.handleHdPathInput.bind(this);
+  }
+
+  componentDidMount() {
+    const { mnemonic, hdpath, passphrase } = this.state;
+    this.attemptWalletDerivation(mnemonic, hdpath, passphrase);
+  }
+
+  handlePassphraseInput(passphrase) {
+    this.setState((state, _) => {
+      const mnemonic = state.mnemonic;
+      const hdpath = state.hdpath;
+      this.attemptWalletDerivation(
+        mnemonic,
+        hdpath === '' ? DEFAULT_HD_PATH : hdpath,
+        passphrase
+      );
+      return {
+        passphrase,
+      };
+    });
+  }
+
+  handleMnemonicInput(mnemonic) {
+    this.setState((state, _) => {
```

Comment:
this code duplication should definitely be its own separate function, but I also already fixed this in the indigo branch, so no need to duplicate the work here. just an fyi